### ISSUE TITLE
When requiresDOM in Test is set to true, unable to find webpack when running tests with Node 8

### ISSUE THREAD

**Issue by Technius** [scalacenter/scalajs-bundler#130]
# Summary

When trying to set up Travis for my project [scalajs-mithril](https://github.com/Technius/scalajs-mithril), I discovered that using the latest version of Node (version 8 at this time) causes webpack to be unable to be found when running tests:

```
[info] Installing jsdom in /opt/project/tests/target/scalajs-bundler-jsdom
[error] npm WARN saveError ENOENT: no such file or directory, open '/opt/project/tests/target/scalajs-bundler-jsdom/package.json'
[error] npm notice created a lockfile as package-lock.json. You should commit this file.
[error] npm WARN enoent ENOENT: no such file or directory, open '/opt/project/tests/target/scalajs-bundler-jsdom/package.json'
[error] npm WARN scalajs-bundler-jsdom No description
[error] npm WARN scalajs-bundler-jsdom No repository field.
[error] npm WARN scalajs-bundler-jsdom No README data
[error] npm WARN scalajs-bundler-jsdom No license field.
[error] 
[info] added 87 packages in 3.208s
[info] Writing and bundling the test loader
[error] module.js:487
[error]     throw err;
[error]     ^
[error] 
[error] Error: Cannot find module '/opt/project/tests/target/scala-2.12/scalajs-bundler/test/node_modules/webpack/bin/webpack'
[error]     at Function.Module._resolveFilename (module.js:485:15)
[error]     at Function.Module._load (module.js:437:25)
[error]     at Function.Module.runMain (module.js:605:10)
[error]     at startup (bootstrap_node.js:158:16)
[error]     at bootstrap_node.js:575:3
```

[Here's the log starting from compilation](https://gist.github.com/Technius/8fe1ab526970ce7c88fb4c7ad0d45fe6).

[Here's the build.sbt](https://github.com/Technius/scalajs-mithril/blob/3d911e86bd5b47d57ea06cc0e461b4a6d03b382e/build.sbt) used for the project.

# Reproducing the Issue

I reproduced the issue by running my project's tests in Docker and changing the version of Node used each time. Here's the Dockerfile that I used to set up the environment:
```dockerfile
FROM debian:testing
RUN apt-get update && apt-get install git curl gnupg openjdk-8-jre-headless apt-transport-https -y
RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 99E82A75642AC823
RUN echo "deb https://sbt.bintray.com/debian /" | tee -a /etc/apt/sources.list
RUN apt-get update && apt-get install sbt -y
RUN curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.2/install.sh | bash
```

I ran the following commands in the container:
```bash
git clone https://github.com/Technius/scalajs-mithril.git /opt/project
cd /opt/project
nvm install 8 # install Node 8
sbt tests/test
```

The issue occurs when using Node 8 (`nvm install 8`) but does not occur when using Node 7 instead (`nvm install 7`).

Later today, I'll come up with a minimal example that reproduces this issue.

**Comment by Technius** [comment]
It seems like the issue happens when `requiresDOM in Test := true`. The minimal example is below:

build.sbt:
```scala
import Dependencies._

lazy val root = (project in file(".")).
  settings(
    inThisBuild(List(
      organization := "com.example",
      scalaVersion := "2.12.1",
      version      := "0.1.0-SNAPSHOT"
    )),
    name := "Hello",
    libraryDependencies += scalaTest % Test,
    requiresDOM in Test := true
  )
  .enablePlugins(ScalaJSPlugin, ScalaJSBundlerPlugin)
```

project/plugins.sbt:
```scala
addSbtPlugin("org.scala-js" % "sbt-scalajs" % "0.6.14")

addSbtPlugin("ch.epfl.scala" % "sbt-scalajs-bundler" % "0.6.0")
```

src/test/scala/example/HelloSpec.scala:
```scala
package example

import org.scalatest._

class HelloSpec extends FlatSpec with Matchers {
  "The Hello object" should "say hello" in {
    Hello.greeting shouldEqual "hello"
  }
}
```

src/main/scala/example/Hello.scala:
```scala
package example

import scala.scalajs.js

object Hello extends Greeting with js.JSApp {
  def main(): Unit = {
    println(greeting)
  }
}

trait Greeting {
  lazy val greeting: String = "hello"
}
```

I put the test project into a folder called `sbtproject` and then used this Dockerfile to set up the environment:
```dockerfile
FROM debian:testing
RUN apt-get update && apt-get install git curl gnupg openjdk-8-jre-headless apt-transport-https -y
RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 99E82A75642AC823
RUN echo "deb https://sbt.bintray.com/debian /" | tee -a /etc/apt/sources.list
RUN apt-get update && apt-get install sbt -y
RUN curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.2/install.sh | bash
ADD sbtproject /opt/project
RUN cd /opt/project && sbt update
```

Then, in the container, I ran `nvm install 8 && cd /opt/project && sbt test` to reproduce the issue. Again, the issue doesn't occur when `nvm install 7` is used instead.

**Comment by julienrf** [comment]
If we downgrade to node < 8 the problem does not happen?

**Comment by helllamer** [comment]
Catched this error today right after `clean; fastOptJS::webpack`.

Issue is related to `npm` 5 (i've tried npm-5.3.0). After downgrading `npm` to 4.6.1, problem gone away.
Plain downgrading of `nodejs` 8 to 7 without `npm` does not help.

Anothen possible workaround is to run `npm install webpack@1.14` inside `target/scala-2.12/scalajs-bundler/main/` directory.

It looks like, npm-5.x ignores `devDependencies` inside `package.json` for some reason.

**Comment by julienrf** [comment]
Do you still have this issue with 0.8.0?

**Comment by helllamer** [comment]
I cannot test against `0.8.x`, because it is not currently possible in my project.

Btw, issue is NOT reproduceable on current `scalajs-bundler` git revision.
Successfully tested with `npm-5.3.0` and `nodejs-8.6.0` on dev and stage builds.

**Comment by julienrf** [comment]
Should we close the issue then?

**Comment by helllamer** [comment]
I think, it may be closed, because nothing to fix here: just to update to 0.9.x.
Dev/prod/tests works ok here.

// I'm not an original issue reporter, so i cannot close.

**Comment by julienrf** [comment]
@Technius I’m closing the issue, feel free to re-open it if needed.
